ansible_collections/arista/avd/roles/eos_designs/templates/l3_edge/logic.j2

```diff
@@ -76,8 +76,8 @@
 {%                 set p2p_bgp_neighbor.peer_group = bgp_peer_groups.IPv4_UNDERLAY_PEERS.name | arista.avd.default("IPv4-UNDERLAY-PEERS") %}
 {%                 set p2p_bgp_neighbor.bfd = p2p_link.bfd | arista.avd.default(
                                               p2p_profile.bfd) %}
-{%                 if switch.bgp_as is arista.avd.defined and as[node_index] != switch.bgp_as %}
-{%                     set p2p_bgp_neighbor.local_as = as[node_index] %}
+{%                 if switch.bgp_as is arista.avd.defined and as[node_index] | string != switch.bgp_as | string %}
```

Comment:
`switch.bgp_as` is set using quotes in the above templates, so you should not need the `| string` filter here on that variable.

Reply:
Removed the | string filter.